### ISSUE TITLE
deprecation warning when setting locale options

### ISSUE THREAD

**Issue by streets-ahead** [moment/moment#3043]
I'm including moment as follows:

```
var moment = require('moment');
moment.locale('en', {
    week: {
        dow: 1 // Monday is the first day of the week.
    }
});
```

As soon as I run this code in Node I get:

```
Deprecation warning: use moment.updateLocale(localeName, config) to change an existing locale. moment.defineLocale(localeName, config) should only be used for creating a new locale
```

I'm using node, but you can also see here: https://jsfiddle.net/6gtctkhj/

It could be that I'm setting the configuration incorrectly, but it seems correct according to [http://momentjs.com/docs/#/customization/](http://momentjs.com/docs/#/customization/).  It looks like the problem might be in the following method:

```
function locale_locales__getSetGlobalLocale (key, values) {
        var data;
        if (key) {
            if (isUndefined(values)) {
                data = locale_locales__getLocale(key);
            }
            else {
                data = defineLocale(key, values);
            }

          ...
        }

        return globalLocale._abbr;
    }
```

This method calls defineLocale, but may need to be updated to call updateLocale?

**Comment by maggiepint** [comment]
Whoops! The docs are bad right there as of 2.12.0 - I'll make a pull request on the docs repository tonight.

See http://momentjs.com/guides/#/warnings/define-locale/
and http://momentjs.com/docs/#/customization/ - the note about 2.12.0

TLDR - just replace .locale with .updateLocale

**Comment by maggiepint** [comment]
Docs updated in moment/momentjs.com#277

**Comment by rijk** [comment]
There are still a lot of references to `moment.locale()` in the [i18n section](http://momentjs.com/docs/#/i18n/changing-locale/) as well.

**Comment by juliomac** [comment]
+1

It does not seem that Docs are updated... lots of references to ``.locale``

I have changed all my ``.locale`` to ``.updateLocale`` and yet got the same message. Possibly one of the packages I use does not know it yet (missing doc info?) or is it not related to the suggested change?

Can;t find a way to get the warning gone... still a warning but... when will it become an error?

**Comment by jomi-se** [comment]
This is an old issue but when I looked for documentation on how to set the `locale` globally through `Google` I ended up in: http://momentjs.com/docs/#/i18n/changing-locale/

Where it says:
```
// From 2.8.1 onward
moment.locale(String);
moment.locale(String[]);
moment.locale(String, Object);

// Deprecated in 2.8.1
moment.lang(String);
moment.lang(String[]);
moment.lang(String, Object);
```
Using `moment.locale(String, Object);`

Gives me the deprecation warning:
```
Deprecation warning: use moment.updateLocale(localeName, config) to change an existing locale. moment.defineLocale(localeName, config) should only be used for creating a new locale
```

**Comment by jmo-jr** [comment]
This issue is still open?! Same message here...

**Comment by silveiratalita** [comment]
Same message here, my problem is that i have a  a screenshot test with jest, and is ok with the dates for while,  cause jest takes the systems date. But tomorrow or in one week these tests will fails, and the screenshots won't be the same.  I tried mock a jest function using  Date function from 'moment-js', but don't work, i don't know why. If someone could help me i'll be glad!

**Comment by toxpal** [comment]
The issue is still here, running the latest version of moment always displays this error in console. Can we expect for a fix?

**Comment by michael-woodward-spok** [comment]
Can this be reopened? It's still a current issue 😦

**Comment by nepa** [comment]
Will this issue ever be fixed? I changed `moment.locale('de')` to `moment.updateLocale('de')`, but the warning persists.

The documentation also references `locale()`, there is only really one section that mentions `updateLocale()`.

I know that the project is considered being finished and will no longer be maintained, but this warning is really annoying.

**Comment by toxpal** [comment]
Since moment.js is in maintenance mode now, I guess it won't be fixed and we will need to move to other libraries slowly.
